src/main/java/de/numcodex/feasibility_gui_backend/query/result/ResultService.java

```diff
@@ -0,0 +1,77 @@
+package de.numcodex.feasibility_gui_backend.query.result;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import de.numcodex.feasibility_gui_backend.query.persistence.ResultType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ResultService {
+
+  @NonNull
+  private Cache<Long, QueryResult> queryResultCache;
+
+  public QueryResult getResult(Long queryId) {
+    return queryResultCache.getIfPresent(queryId);
+  }
+
+  @Synchronized
+  public void addResultLine(Long queryId, ResultLine resultLine) {
+    String siteName = resultLine.getSiteName();
+    QueryResult queryResult = queryResultCache.getIfPresent(queryId);
+    if (queryResult == null) {
+      queryResult = new QueryResult();
+    } else if (queryResult.getResultsBySite().containsKey(siteName)) {
+      // Only store one result per site - new results get discarded
+      return;
+    }
+
+    if (queryResult.getTotalResult() == null) {
+      queryResult.setTotalResult(resultLine.getResult());
+    } else if (resultLine.getResult() == null){
+      queryResult.setTotalResult(queryResult.getTotalResult());
+    } else {
+      queryResult.setTotalResult(queryResult.getTotalResult() + resultLine.getResult());
+    }
+    queryResult.getResultsBySite().put(siteName, resultLine);
+    queryResultCache.put(queryId, queryResult);
+  }
+
+  @Synchronized
+  public List<ResultLine> findByQueryAndStatus(long queryId, ResultType resultType) {
+    List<ResultLine> resultLines = new ArrayList<>();
+    QueryResult queryResult = queryResultCache.getIfPresent(queryId);
+
+    if (queryResult != null) {
+      for (Entry<String, ResultLine> resultLine : queryResult.getResultsBySite().entrySet()) {
+        if (resultLine.getValue().getResultType() == resultType) {
+          resultLines.add(resultLine.getValue());
+        }
+      }
+    }
+
+    return resultLines;
+  }
+
+  public long count() {
+    return queryResultCache.estimatedSize();
+  }
+
+  public List<QueryResult> findAll() {
```

Comment:
This method is also only used in tests, so please think about if you really need it for the tests and if so please test it also in the `ResultServiceTest`.